ApixMiddleware.php

```diff
@@ -42,15 +45,28 @@ public function __construct(
      * @param \Symfony\Component\HttpFoundation\Request $request
      * @return \Psr\Http\Message\ResponseInterface
      */
-    public function before(Request $request)
+    public function before(RequestEvent $event)
     {
+
+        if (!$event->isMasterRequest()) {
+            return;
+        }
```

Comment:
Everytime I look at this code I wonder why I did this. I think it was to avoid executing multiple times on a single request. But I can try to figure it out again and better document it.

Reply:
So looking back in here, I am correct that this was to avoid running the code on sub-requests and only on the initial master request.

Reply:
Well, yes... but... I guess it's not obvious _where_ the multiple requests are happening? As in, is it not:

1. Crayfish app kernel receives the request, generates the request event to be routed
2. This subscriber does its thing with headers/attributes/whatever
3. The controller finally receives the request, and acts on it

... for it to be an issue, the controller itself would have to be asking the kernel to perform a subrequest, yeah?

Reply:
I'll try pulling it back out and see if I can see exactly why I added it in.

Reply:
Removing this seemed to have no real effect so I have removed it.